src/suggestion-box.ts

```diff
@@ -0,0 +1,383 @@
+import { LitElement, html, css } from "lit";
+import { customElement, property, state } from "lit/decorators.js";
+import { queryAssignedElements } from "lit/decorators.js";
+
+@customElement("aalam-sgn-box")
+export class SuggestionBox extends LitElement {
+    static styles = css`
+        :host {
+            display: block;
+            position: relative;
+        }
+    `;
+
+    @property({ type: Array<Object> })
+    list: Array<{ [key: string]: any }> = [];
+
+    @property()
+    listkey: string = "";
+
+    @property()
+    slot_input = null;
+
+    @property()
+    _private_slot_el: boolean = false;
+
+    @queryAssignedElements({ slot: "sgn-item-template" })
+    template_slots: Array<HTMLElement>;
+
+    @queryAssignedElements({ slot: "__private-item" })
+    private_items: Array<HTMLElement>;
+
+    @property()
+    prevItem: number = 0;
+
+    @state()
+    has_more: boolean = false;
+
+    @state()
+    index: number = -1;
+
+    @property()
+    size: number = 0;
+
+    @state()
+    show_empty: boolean = true;
+
+    @property()
+    result: Array<{ [key: string]: string }> = [];
+
+    @property()
+    highlight: string = "";
+
+    @property()
+    curr_index: number = 0;
+
+    @property()
+    input_el: HTMLElement | null = null;
+
+    @property()
+    filtered_list: Array<{ [key: string]: string }> = [];
+
+    @property()
+    prevIndex: number = 0;
+
+    @state()
+    show_container = false;
+
+    private _outClickListener = this.windowClickEvent.bind(this);
+    private input_temp: HTMLElement | null | string = null;
+
+    connectedCallback() {
+        super.connectedCallback();
+        this.addEventListener("mouseout", this._mouseOutEvent);
+        document.addEventListener("click", this._outClickListener);
+    }
+    disconnectedCallback() {
+        super.disconnectedCallback();
+        this.removeEventListener("mouseout", this._mouseOutEvent);
+        document.addEventListener("click", this._outClickListener);
+    }
+    render() {
+        return html`
+            <div style="position:relative">
+                <div
+                    part="sgn-input"
+                    class="inputbox"
+                    @focusin=${this._inputFocusEvent}
+                    @keydown=${this.keyDownInputEvent}
+                >
```

Comment:
Its better not to have the > hanging. It would be better if it ends in the previous line.